models/products/product.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const Model = require('../model.js');
+
+const schema = {
+    fields: {
+        id: { type: 'string', required: true },
+        category_id: { type: 'string', required: true },
+        price: { type: 'number', required: true },
+        weight: { type: 'number' },
+        quantity_in_stock: { type: 'number', required: true },
+    }
+};
+
+class Product extends Model {
```

Comment:
it should extends to the MemCollection class(or memory-data-model.js) because that has the create() function, right now it going to the model.js file  to create the new data.

Reply:
In my implementation, Products extends to MemCollection, and Product would be passed into Products for schema verification. I think there are multiple ways to solve this.